pkg/generators/extension.go

```diff
@@ -63,17 +63,25 @@ type extension struct {
 	values []string // Example: [atomic]
 }
 
+func (e extension) hasAllowedValues() bool {
```

Comment:
I think the relevant detail is whether the allowed values are known ahead of time or not, not whether there are allowed values. Tricky to think of a better name. [just a comment, no action requested]